### DISCUSSION TITLE
Different ways of calling useStore.  How do they differ?

### DISCUSSION THREAD

**Discussion by HarperTime** [pmndrs/zustand#2447] [Q&A] [answered]
Are these these access methods equivalent:?  For each, when will updates be triggered?

```
1. const { item} = useItemStore( (state) => state) )
2. const item = useItemStore.getState().item
3. const item = useItemStore().item
```

**Comment by dbritto-dev** [comment, marked as answer]
@HarperTime 1 and 3 are the same but 2 wouldn't trigger new renders

**Reply by HarperTime** [reply]
Looking at "Fetching Everything" in the docs.  It looks like 1 and 3 would trigger on any state change, not just item.  Right?

**Reply by dbritto-dev** [reply]
@HarperTime yes

**Reply by dbritto-dev** [reply]
btw, using selectors for primitive data types are OK but anything else you need to use equality fn like `shallow`

**Reply by HarperTime** [reply]
Really glad you added that.  So, say i have a method on the store getUnsoldItems that returns an array of objects:
```
a. const { unsoldItems } = useItemStore((state) => getUnsoldItems())
b. const { unsoldItems } = useItemStore(useShallow((state) => getUnsoldItems()))
```

How would redraw behavior be different between version a and version b when the return of the function changes (adding items to array, changing a value within an element of the array)?

**Reply by dbritto-dev** [reply]
@HarperTime talking about arrays you need use a deep comparison
